src/generators/legacy-html/utils/safeCopy.mjs

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+import { readFile, writeFile, stat, readdir } from 'node:fs/promises';
+import { join } from 'node:path';
+
+/**
+ * Safely copies files from source to target directory, skipping files that haven't changed
+ * based on file stats (size and modification time)
+ *
+ * @param {string} srcDir - Source directory path
+ * @param {string} targetDir - Target directory path
+ */
+export async function safeCopy(srcDir, targetDir) {
+  const files = await readdir(srcDir);
+
+  for (const file of files) {
+    const sourcePath = join(srcDir, file);
+    const targetPath = join(targetDir, file);
+
+    const [sStat, tStat] = await Promise.allSettled([
+      stat(sourcePath),
+      stat(targetPath),
+    ]);
+
+    const shouldWrite =
+      tStat.status === 'rejected' ||
+      sStat.value.size !== tStat.value.size ||
+      sStat.value.mtimeMs > tStat.value.mtimeMs;
+
+    if (!shouldWrite) {
+      continue;
+    }
+
+    const fileContent = await readFile(sourcePath);
+
+    await writeFile(targetPath, fileContent);
```

Comment:
This is not "safe", there's still a TOCTOU: you call `stat` without locking the file, so when `writeFile` is reached the `shouldWrite` can be outdated (and with several processes running concurrently, it's almost guaranteed to happen).

Reply:
Yeah, this PR doesn't prevent concurrent reads/writes, it just "prevents" the fs issues you were facing. It's safe in the sense that it shouldn't fail due to concurrency issues, or very unlikely.